conjurapi/authn_test.go

```diff
@@ -151,89 +152,112 @@ func runRotateUserAPIKeyAssertions(t *testing.T, tc rotateUserAPIKeyTestCase, co
 	assert.NoError(t, err)
 }
 
-func TestClient_OidcAuthenticate(t *testing.T) {
-	testCases := []struct {
-		name                 string
-		tokenPath            string
-		expectFileWriteError bool
-	}{
-		{
-			name:      "Caches token to default location",
-			tokenPath: "",
-		},
-		{
-			name:      "Caches token to custom location",
-			tokenPath: t.TempDir() + "/tmp/test-token",
-		},
-		{
-			// Writing this file will fail but there should be no error returned from OidcAuthenticate()
-			name:                 "Caches token to custom location with trailing slash",
-			tokenPath:            t.TempDir() + "/tmp/test-token/",
-			expectFileWriteError: true,
-		},
-	}
+func TestClient_Login(t *testing.T) {
+	t.Run("Login and Authenticate", func(t *testing.T) {
+		ts, client := setupTestClient(t)
+		defer ts.Close()
+
+		token, err := client.Login("alice", "password")
+		assert.NoError(t, err)
+		assert.Equal(t, "test-api-key", string(token))
+
+		// Check that api key was cached to the correct location
+		contents, err := os.ReadFile(client.GetConfig().NetRCPath)
+		assert.NoError(t, err)
```

Comment:
nit: It feels like this is double-testing the storeCredentials functionality instead of just asserting that `storeCredentials` was called with the correct params (same below for OidcAuthenticate)

Reply:
This is a great point. In the next part of this effort we'll be adding support for other storage backings, which will make it easy to mock this out. I'd prefer to merge this as is for now and change that abstraction in the next effort.